### ISSUE TITLE
Should not get hints within same persona

### ISSUE THREAD

**Issue by fkleedorfer** [researchstudio-sat/webofneeds#3007]
It is not clear how this situation should be prevented in principle, but it is a UX problem to receive a normal chatSocket hint to an atom held by the same persona.

![grafik](https://user-images.githubusercontent.com/551624/61578089-99760680-aaf1-11e9-8831-6b8a8faf331a.png)

Disallowing hints between atoms with the same holder seems a little arbitrary: 
* maybe other external relationships (other than the holder) require such behaviour. 
* maybe in some situations, matching between one's own atoms is quite useful - probably not the chatSocket, through.

Disallowing this hint as an additional rule defined by the socket might make sense; this could be a standard rule that sockets enable by default.

**Comment by quasarchimaere** [comment]
i agree, matchers should not produce hints for certain socket types with the same holder.
